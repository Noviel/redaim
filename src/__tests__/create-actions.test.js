import {  
  createActions,
  createActionsR,
} from '../actions';

describe('createActions', () => {
  it('should create actions from object with objects as values', () => {
    const actionsObject = {
      exec1: { exec: 'EXEC1', payload: { value: 1 } },
      exec2: { exec: 'EXEC2', payload: { value: 2 } },
      exec3: { exec: 'EXEC3', payload: { value: 3 } }
    };

    const { exec1, exec2, exec3 } = createActions('TRIGGER')('target')(actionsObject);

    const result = {
      exec1: exec1(),
      exec2: exec2(),
      exec3: exec3()
    };

    expect(result).toMatchSnapshot();
  });

  it('should create actions from object with functions as values', () => {
    const actionsObject = {
      exec1: value => ({ exec: 'EXEC1', payload: { value } }),
      exec2: (value1, value2) => ({ exec: 'EXEC2', payload: { value1, value2 } }),
    };

    const { exec1, exec2 } = createActions('TRIGGER')('target')(actionsObject);

    const result = {
      exec1: exec1(100),
      exec2: exec2({ value: 'value'}, 420),
    };

    expect(result).toMatchSnapshot();
  });

  it('should create actions from object with mixed values', () => {
    const actionsObject = {
      exec1: value => ({ exec: 'EXEC1', payload: { value } }),
      exec2: { exec: 'EXEC2', payload: 'payload' },
    };

    const { exec1, exec2 } = createActions('TRIGGER')('target')(actionsObject);

    const result = {
      exec1: exec1('exec1-value'),
      exec2: exec2(),
    };

    expect(result).toMatchSnapshot();
  });
});

describe('createActionsR', () => {
  it('should create actions from object with object as values', () => {
    const actionsObject = {
      exec1: { exec: 'EXEC1', payload: { value: 1 } },
      exec2: { exec: 'EXEC2', payload: { value: 2 } },
      exec3: { exec: 'EXEC3', payload: { value: 3 } }
    };

    const { exec1, exec2, exec3 } = createActionsR(actionsObject)('TRIGGER')('target');

    const result = {
      exec1: exec1(),
      exec2: exec2(),
      exec3: exec3()
    };

    expect(result).toMatchSnapshot();
  });

  it('should create actions from object with functions as values', () => {
    const actionsObject = {
      exec1: value => ({ exec: 'EXEC1', payload: { value } }),
      exec2: (value1, value2) => ({ exec: 'EXEC2', payload: { value1, value2 } }),
    };

    const { exec1, exec2 } = createActionsR(actionsObject)('TRIGGER')('target');

    const result = {
      exec1: exec1(100),
      exec2: exec2({ value: 'value'}, 420),
    };

    expect(result).toMatchSnapshot();
  });

  it('should create actions from object with mixed values', () => {
    const actionsObject = {
      exec1: value => ({ exec: 'EXEC1', payload: { value } }),
      exec2: { exec: 'EXEC2', payload: 'payload' },
    };

    const { exec1, exec2 } = createActionsR(actionsObject)('TRIGGER')('target');

    const result = {
      exec1: exec1('exec1-value'),
      exec2: exec2(),
    };

    expect(result).toMatchSnapshot();
  });
});
