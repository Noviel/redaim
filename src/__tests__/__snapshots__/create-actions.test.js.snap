// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createActions should create actions from object with functions as values 1`] = `
Object {
  "exec1": Object {
    "meta": Object {
      "exec": "EXEC1",
      "target": "target",
    },
    "payload": Object {
      "value": 100,
    },
    "type": "TRIGGER",
  },
  "exec2": Object {
    "meta": Object {
      "exec": "EXEC2",
      "target": "target",
    },
    "payload": Object {
      "value1": Object {
        "value": "value",
      },
      "value2": 420,
    },
    "type": "TRIGGER",
  },
}
`;

exports[`createActions should create actions from object with mixed values 1`] = `
Object {
  "exec1": Object {
    "meta": Object {
      "exec": "EXEC1",
      "target": "target",
    },
    "payload": Object {
      "value": "exec1-value",
    },
    "type": "TRIGGER",
  },
  "exec2": Object {
    "meta": Object {
      "exec": "EXEC2",
      "target": "target",
    },
    "payload": "payload",
    "type": "TRIGGER",
  },
}
`;

exports[`createActions should create actions from object with objects as values 1`] = `
Object {
  "exec1": Object {
    "meta": Object {
      "exec": "EXEC1",
      "target": "target",
    },
    "payload": Object {
      "value": 1,
    },
    "type": "TRIGGER",
  },
  "exec2": Object {
    "meta": Object {
      "exec": "EXEC2",
      "target": "target",
    },
    "payload": Object {
      "value": 2,
    },
    "type": "TRIGGER",
  },
  "exec3": Object {
    "meta": Object {
      "exec": "EXEC3",
      "target": "target",
    },
    "payload": Object {
      "value": 3,
    },
    "type": "TRIGGER",
  },
}
`;

exports[`createActionsR should create actions from object with functions as values 1`] = `
Object {
  "exec1": Object {
    "meta": Object {
      "exec": "EXEC1",
      "target": "target",
    },
    "payload": Object {
      "value": 100,
    },
    "type": "TRIGGER",
  },
  "exec2": Object {
    "meta": Object {
      "exec": "EXEC2",
      "target": "target",
    },
    "payload": Object {
      "value1": Object {
        "value": "value",
      },
      "value2": 420,
    },
    "type": "TRIGGER",
  },
}
`;

exports[`createActionsR should create actions from object with mixed values 1`] = `
Object {
  "exec1": Object {
    "meta": Object {
      "exec": "EXEC1",
      "target": "target",
    },
    "payload": Object {
      "value": "exec1-value",
    },
    "type": "TRIGGER",
  },
  "exec2": Object {
    "meta": Object {
      "exec": "EXEC2",
      "target": "target",
    },
    "payload": "payload",
    "type": "TRIGGER",
  },
}
`;

exports[`createActionsR should create actions from object with object as values 1`] = `
Object {
  "exec1": Object {
    "meta": Object {
      "exec": "EXEC1",
      "target": "target",
    },
    "payload": Object {
      "value": 1,
    },
    "type": "TRIGGER",
  },
  "exec2": Object {
    "meta": Object {
      "exec": "EXEC2",
      "target": "target",
    },
    "payload": Object {
      "value": 2,
    },
    "type": "TRIGGER",
  },
  "exec3": Object {
    "meta": Object {
      "exec": "EXEC3",
      "target": "target",
    },
    "payload": Object {
      "value": 3,
    },
    "type": "TRIGGER",
  },
}
`;
